### PR TITLE
Fix interface status missing when creating a device with a device role associated to a InterfaceTemplate

### PR DESCRIPTION
**Pull request by timizuoebideri1**
# Closes: #2419 
# What's Changed

# TODO

- [ ] Explanation of Change(s)
- [ ] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [ ] Attached Screenshots, Payload Example
- [ ] Unit, Integration Tests
- [ ] Documentation Updates (when adding/changing features)
- [ ] Example Plugin Updates (when adding/changing features)
- [ ] Outline Remaining Work, Constraints from Design

### CODE REVIEW

File: nautobot/dcim/models/device_component_templates.py [hunk 1]
```diff
@@ -258,11 +259,16 @@ class Meta:
         unique_together = ("device_type", "name")
 
     def instantiate(self, device):
+        try:
+            status = Status.objects.get_for_model(Interface).get(slug=InterfaceStatusChoices.STATUS_ACTIVE)
+        except Status.DoesNotExist:
+            status = Status.objects.get_for_model(Interface).first()
```
Comment (glennmatthews) [inline nautobot/dcim/models/device_component_templates.py:265]: I know we're moving away from `cacheops` - I wonder if we should cache this in memory somewhere? When instantiating a DeviceType with a couple hundred InterfaceTemplates, is this going to hit the database each time?

File: nautobot/dcim/tests/test_models.py [hunk 2]
```diff
@@ -147,6 +147,34 @@ def test_instantiate_model(self):
         self.assertEqual(Interface.objects.get(pk=interfaces[1].pk).cf["field_3"], "value_3")
 
 
+class InterfaceTemplateTestCase(TestCase):
```
Comment (gsnider2195) [inline nautobot/dcim/tests/test_models.py:150]: Can you add a test for when the "active" status doesn't exist?

File: nautobot/dcim/models/device_component_templates.py (hunk 1 above)
Comment (timizuoebideri1) [reply nautobot/dcim/models/device_component_templates.py:265]: Do you recommend that I use an alternative memory cache, such as `django.core.cache.backends.locmem.LocMemCache`?
Or i add a TODO for this

Comment (glennmatthews) [reply nautobot/dcim/models/device_component_templates.py:265]: I wasn't thinking anything so fancy - I was thinking more of an approach like we have for `Prefix.STATUS_CONTAINER` where we just look it up and store it on the class for future re-use.

**Review by glennmatthews — APPROVED** [review]

**Review by jathanism — APPROVED** [review]

**Pull request merged**